src/maximal_cliques_template.h

```diff
@@ -100,6 +104,39 @@
 } while (0)
 #endif
 
+#ifdef IGRAPH_MC_NUMBER
+#define RESTYPE igraph_integer_t *res
+#define RESNAME res
+#define SUFFIX _number
+#define PREFIX i_
+#define RECORD do { \
+  if (*res < clsize) \
+    *res = clsize; \
+  } while (0)
+#define FINALLY *res=0;
+#define FOR_LOOP_OVER_VERTICES for (i=0; i<no_of_nodes; i++) {
+#define FOR_LOOP_OVER_VERTICES_PREPARE
+#endif
+
+#ifdef IGRAPH_MC_SINGLE_LARGEST
+#define RESTYPE igraph_vector_t *res
+#define RESNAME res
+#define SUFFIX _single_largest
+#define PREFIX i_
+#define RECORD do { \
+  if (igraph_vector_size(res) < clsize){ \
+    igraph_vector_init(res, clsize); \
```

Comment:
Isn't this a memory leak?  It would seem that a new clique vector is allocated every time a larger clique is found, but the old one is never freed.  It's possible that I am misunderstanding something.